### PR TITLE
Enum tests

### PR DESCRIPTION
**Pull request by davfsa**
- Remove unreachable and duplicate code
   - Fixed some logic

### Checklist

- [x] I have run `nox` and all the pipelines have passed.
- [x] I have made unittests according to the code I have added/modified/deleted.

### Related issues
#47

### CODE REVIEW

File: hikari/internal/enums.py [hunk 1]
```diff
@@ -122,8 +109,15 @@ def __call__(cls, value: typing.Any) -> typing.Any:
...
         return cls._name_to_member_map_[name]
 
+    def __contains__(cls, item: typing.Any) -> bool:
+        try:
+            cls._value_to_member_map_[item]
+            return True
+        except KeyError:
+            return False
+
     def __iter__(cls) -> typing.Iterator[str]:
```
Comment (FasterSpeeding) [inline hikari/internal/enums.py:119]: would iterating values here not make it an iterator of Any not str

File: hikari/internal/enums.py [hunk 2]
```diff
@@ -122,8 +109,15 @@ def __call__(cls, value: typing.Any) -> typing.Any:
     def __getitem__(cls, name: str) -> typing.Any:
         return cls._name_to_member_map_[name]
 
+    def __contains__(cls, item: typing.Any) -> bool:
+        try:
+            cls._value_to_member_map_[item]
```
Comment (FasterSpeeding) [inline hikari/internal/enums.py:114]: how does this compare to just returning `item in cls._value_to_member_map_`?

Comment (davfsa) [reply hikari/internal/enums.py:114]: Same thing, just one call less, but will change if it makes it cleaner

File: hikari/internal/enums.py (hunk 1 above)
Comment (davfsa) [reply hikari/internal/enums.py:119]: Yep, will fix typehint

**Comment by davfsa** [comment]
@FasterSpeeding all fixed

**Review by FasterSpeeding — APPROVED** [review]

**Pull request merged**